packages/api/src/middleware/auth.js

```diff
@@ -11,31 +19,37 @@ function authFactory(params) {
   return async (req, res, next) => {
     // must have either an API key (starts with 'Bearer') or a JWT token
     const authToken = req.headers.authorization;
+    const { tokenType, tokenValue } = parseAuthToken(authToken);
+    const basicUser = basicAuth.parse(authToken);
     let user;
     let tokenObject;
     let userId;
 
-    if (!authToken) {
-      throw new ForbiddenError(`no token object ${authToken} found`);
-    } else if (authToken.startsWith("Bearer")) {
-      tokenObject = await req.store.get(`api-token/${req.token}`);
-      if (!tokenObject) {
-        throw new ForbiddenError(`no token object ${authToken} found`);
+    if (!tokenType) {
+      throw new ForbiddenError(`no authorization header provided`);
+    } else if (["Bearer", "Basic"].includes(tokenType)) {
+      const isBasic = tokenType === "Basic";
+      tokenObject = await db.apiToken.get(isBasic ? basicUser.pass : req.token);
```

Comment:
`basicUser.pass ` - this assumes that `basicUser` was successfully parsed, which not necessary be the case.

Reply:
Good catch. I could swear I read somewhere it returned an empty object when the parsing failed, but apparently it's not the case.

Reply:
Empty object will not work too - then `db.apiToken.get` will throw error because it expect id to be provided.

Reply:
I see. Alright so I can't even coalesce to an empty string here, I'll do an explicit check.

Reply:
While we're at it, do you know where this `req.token` property comes from? I coudn't find it on express docs nor any code of ours setting it explicitly.

Reply:
Nevermind just found it: https://github.com/livepeer/livepeer-com/blob/d903f4e5761b05bd7469c925c8a4201f558ca38e/packages/api/src/app-router.js#L120

Maybe I'll use the [`express-basic-auth`](https://www.npmjs.com/package/express-basic-auth) instead of this more manual `basic-auth` then, for consistency.

Reply:
Fixed. I'll check again later if it's worth migrating to the `express-basic-auth`, it seems to do way more than just parsing the headers so it may be overkill.